### PR TITLE
First attempt at logging to files instead of std out

### PR DESCRIPTION
**Pull request by The1Penguin**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -37,6 +37,7 @@ import           System.Console.GetOpt                    ( ArgDescr(..)
                                                           , getOpt
                                                           , usageInfo
                                                           )
+import           System.Directory                         ( createDirectoryIfMissing )
```
Comment (Rembane) [inline app/Main.hs:40]: :tada:  IT EXISTS FOR REAL! :tada:

File: app/Main.hs [hunk 2]
```diff
@@ -54,6 +55,7 @@ opts :: [OptDescr (Config -> Config)]
 opts =
   [ Option [] ["help"] (NoArg (set cHelp True))           "Show usage info"
   , Option [] ["port"] (ReqArg (set cPort . read) "PORT") "Port to run on"
+  , Option [] ["path"] (ReqArg (set cLogPath) "PATH") "Path to save log files to, default is 'logs'"
```
Comment (Rembane) [inline app/Main.hs]:
```suggestion
  , Option [] ["path"] (ReqArg (set cLogPath) "PATH")     "Path to save log files to, default is 'logs'"
```

Columns <3

File: app/Main.hs [hunk 3]
```diff
@@ -71,6 +73,7 @@ main = (recreateConfig . getOpt Permute opts <$> getArgs) >>= \case
   (config                  , _    , _    ) -> do
     upd     <- newEmptyMVar -- putMVar when to update
     viewRef <- createViewReference
+    createDirectoryIfMissing True (_cLogPath config)
```
Comment (Rembane) [inline app/Main.hs:76]: Is this path relative?

Comment (Rembane) [reply app/Main.hs:76]: Or maybe both?

Comment (Rembane) [reply app/Main.hs:76]: Both absolute and relative that is

**Review by SwedishSubmarine — APPROVED** [review]
Looks good to me. Not sure how to unit test this but shouldn't break anything either way so worth trying.

File: src/Model.hs [hunk 4]
```diff
@@ -31,18 +31,14 @@ import           Data.IORef                               ( IORef
...
-                                                          )
+import           Data.Text.Lazy                           ( fromStrict )
+import           Prettyprinter                            ( Doc )
 import           Data.AffineSpace                         ( (.+^) )
 import           Data.Thyme                               ( _localDay
                                                           , _localTimeOfDay
                                                           , _todHour
                                                           , _zonedTimeToLocalTime
-                                                          , getZonedTime
+                                                          , getZonedTime, getCurrentTime
```
Comment (Rembane) [inline src/Model.hs]:
```suggestion
                                                          , getCurrentTime
                                                          , getZonedTime
```

Comment (Rembane) [reply src/Model.hs]: Newlines and lexicographic ordering <3

**Review by Rembane — CHANGES_REQUESTED** [review]
Damn good initiative if I may say so myself!

See the comments for confusion, hearts and questions!

File: src/Model.hs [hunk 5]
```diff
@@ -56,6 +52,7 @@ import           Model.Types
 import           Model.Karen
 import           Model.Wijkanders
 import           Model.Linsen
+import Text.Printf (printf)
```
Comment (Multipacker) [inline src/Model.hs]: Minor indentation fix
```suggestion
import           Model.Linsen
import           Text.Printf (printf)
```

**Review by Multipacker — APPROVED** [review]
Looks good, just a minor indentation fix

File: app/Main.hs (hunk 3 above)
Comment (The1Penguin) [reply app/Main.hs:76]: It handles both :D :dagger:

Comment (Rembane) [reply app/Main.hs:76]: AWESOME! I am now a much calmer human.

**Comment by Rembane** [comment]
Something just struck me, this is me thinking aloud, I think it's faster and uses less resources to write to file as a `ByteString` or `Text`, OTOH I think we'll have to go through `String` anyway, so maybe it doesn't matter?

**Comment by The1Penguin** [comment]
> Something just struck me, this is me thinking aloud, I think it's faster and uses less resources to write to file as a `ByteString` or `Text`, OTOH I think we'll have to go through `String` anyway, so maybe it doesn't matter?

Ignored :dagger:

**Review by Rembane — APPROVED** [review]
I've done art! You can use it if you want to. :smile_cat:

**Pull request merged**